Rakefile

```diff
@@ -114,6 +116,31 @@ namespace :gem do
     end
   end
 
+  desc "document all gems"
+  task docs: [:girdocs, :rubydocs]
```

Comment:
I'm OK with separated docs as the first step.
But I want to mix `yard-gobject-introspection` based documents and `*.rb`/`.h`/`*.c` based documents eventually.

Reply:
It might be possible for yard-gobject-introspection to be used a plugin for yard?

I was looking recently at the structure of the Invoker, FunctionInfo and virtual function structures in loader.rb for gobject-introspection. Perhaps these could be used for merging the gobject introspection metadata and its implementation in Ruby into the YRI metadata structures used in Yard. 

I believe that Yard's YRI data may be similar to the RI structures in rdoc. Ostensibly, this YRI data could be used directly for documentation browser for the desktop, as well as for any role in producing a merged HTML documentation. I'll try to take a look at the Yard API here.

First, to update the patch for the Rakefile though